In [1]:
import pandas as pd 
import numpy as np 
from scipy import stats 

In [2]:
df = pd.read_csv('../data/retail_sales.csv')

In [6]:
# hypothesis and cofidence intrevals 
#two groups 
male_spend = df[df["Gender"]=='Male']['Total Amount']
female_spend= df[df["Gender"]=='Female']['Total Amount']

print("male avg spend", male_spend.mean().round(2))
print("female avg spend",female_spend.mean().round(2))
#t-test

t_stat, p_value = stats.ttest_ind(male_spend, female_spend)

print(f"\n t-statistics: {t_stat:.4f}")
print(f"p-value: {p_value:.4f}")

if(p_value<0.05):
        print("reject h0: spending is significantly differnt")
else: 
        print("keep h0: no significant difference in spending")

#chi square test
# testing product are diff fro genders ? 

contingency = pd.crosstab(df['Gender'], df['Product Category'])
print("\ncontigency table:")
print(contingency)

chi2, p_chi, dof , expected = stats.chi2_contingency(contingency)
print(f"\nchi-square p-value: {p_chi:.4f}")

if p_chi<0.05:
        print("gender and product are related")
else: 
        print("no significant relationship between gender and category")

#confidence intervals 
mean_spend = df['Total Amount'].mean()
ci= stats.t.interval(
        confidence= 0.95, 
        df= len(df)-1,
        loc=mean_spend,
        scale= stats.sem(df['Total Amount'])
)

print(f"\n Mean spend: {mean_spend:.2f}")
print(f"95% ci : {ci[0]:.2f} to {ci[1]:.2f}")

male avg spend 455.43
female avg spend 456.55

 t-statistics: -0.0316
p-value: 0.9748
keep h0: no significant difference in spending

contigency table:
Product Category  Beauty  Clothing  Electronics
Gender                                         
Female               166       174          170
Male                 141       177          172

chi-square p-value: 0.4330
no significant relationship between gender and category

 Mean spend: 456.00
95% ci : 421.25 to 490.75



Hypothesis: Males and females spend differently
Result: REJECTED — p-value 0.97, difference only ₹1.12
Insight: Gender is not a useful segment for spend analysis

Hypothesis: Gender affects product category preference  
Result: REJECTED — p-value 0.43
Insight: All genders buy all categories equally
